### DISCUSSION TITLE
From entropy, BIP84

### DISCUSSION THREAD

**Discussion by pbies** [hdwallet-io/python-hdwallet#110] [Q&A] [answered]
Hi!

Need help with getting BIP84 HD wallet from entropy.

For now I don't see from_entropy function in new, 3.2.0 version with BIP84.

Do I need to go back with HDWallet version, or it can be done in 3.2.0?

Thank you

**Comment by meherett** [comment, marked as answer]
Hello @pbies,

In Version 3, the `from_entropy` and `from_mnemonic` root functions are not available in all HD classes. Instead, the root function starts with `from_seed`. To use entropy or a mnemonic, you need to convert them into a seed first.

For further details: https://github.com/talonlab/python-hdwallet/blob/master/clients/bip84.py#L50

Alternatively, if you prefer to use `from_entropy` and `from_mnemonic` directly, consider using the `HDWallet` class instead of the specific HD classes.

See this example: https://github.com/talonlab/python-hdwallet/blob/master/examples/BIPs/from_entropy.py

Thanks!

**Comment by pbies** [comment]
Which BIPs/BIPxxHDs are the default ones for Bitcoin? I see: 84, 39, 44, 32.
I need from_entropy, from_seed and from_private_key for standard Bitcoins. The ones used in Bitcoin Core.
I will be converting to all 7 address types.
In which places use which BIPs?

I have code like:
```
from hdwallet import HDWallet
from hdwallet.cryptocurrencies import Bitcoin as BTC
from hdwallet.entropies import BIP39Entropy
from hdwallet.hds import BIP84HD
from hdwallet.mnemonics import BIP39Mnemonic
from hdwallet.seeds import BIP39Seed

hdwallet=HDWallet(cryptocurrency=BTC, hd=BIP84HD).from_mnemonic(mnemonic=BIP39Mnemonic(mnemonic=k))
hdwallet=HDWallet(cryptocurrency=BTC, hd=BIP84HD).from_private_key(private_key=k)
hdwallet=HDWallet(cryptocurrency=BTC, hd=BIP84HD).from_seed(seed=BIP39Seed(k))
hdwallet=HDWallet(cryptocurrency=BTC, hd=BIP84HD).from_wif(wif=k)
hdwallet=HDWallet(cryptocurrency=BTC, hd=BIP84HD).from_xprivate_key(xprivate_key=k)
hdwallet=HDWallet(cryptocurrency=BTC, hd=BIP84HD).from_entropy(entropy=BIP39Entropy(k))
```
and then printing addresses from hdwallet.

**Comment by meherett** [comment]
Bitcoin Core typically defaults to the following address types:

- Legacy Addresses (`P2PKH`): Based on `BIP44HD`, though these are now seldom used due to inefficiency.
- Nested SegWit Addresses (`P2WPKH-In-P2SH`): Aligned with `BIP49HD`, offering compatibility with older systems.
- Native SegWit (`P2WPKH`): Defined by `BIP84HD`, this is the preferred format for modern wallets.
- Taproot Addresses (`P2TR`): Specified by `BIP86HD`, used for advanced functionalities like complex smart contracts.

If you aim to convert to all 7 address types, you can use `BIP32HD`. However, this is not the default approach used by Bitcoin Core.

**Reply by pbies** [reply]
Thank you very much!
